pkgs/misc/emulators/mame/default.nix

```diff
@@ -34,17 +47,16 @@ in mkDerivation {
     "TOOLS=1"
     "USE_LIBSDL=1"
   ]
-  ++ lib.optionals stdenv.cc.isClang [ "CC=clang" "CXX=clang++" ]
-  ;
+  ++ lib.optionals stdenv.cc.isClang [ "CC=clang" "CXX=clang++" ];
```

Comment:
how about unconditionally providing `CC=${stdenv.cc.targetPrefix}cc CXX=${stdenv.cc.targetPrefix}c++`

Reply:
I don't think this would work:

```
nix-repl> stdenv.cc.targetPrefix
""
```

Reply:
Yeah that's right for the native case. Does the makefile not like `CC=cc`?

Reply:
Got this on macOS:

```
nix-repl> stdenv.cc.targetPrefix
""
```

Or unless this is something that is only changed on `stdenv.mkDerivation` :thinking: ?

Reply:
Yeah that's correct. Whenever not cross-compiling, targetPrefix is an empty string. The point of providing it to gracefully handle cross-compilation scenarios; e.g. when cross-compiling to i686-linux, targetPrefix will be "i686-unknown-linux-gnu-".

TLDR ${stdenv.cc.targetPrefix}cc is the most generic way to provide the name of the C compiler, and should work in all cases

Reply:
Ok, replaced the old CC/CXX flags with those ones.

Let me test if this will work on macOS...

Reply:
Got the following error on macOS:

http://ix.io/3FDL

Strange, I thought our default Clang was 7.0 :thinking: .

Reply:
It seems to be using Clang 5.0.

Can I force Clang 6.0 or 7.0 without changing to `clangStdenv` :thinking: ? Only for macOS obviously.

Reply:
Whoa that's weird, I can't find where that's override happens. The clang5 override has to be somewhere, our stdenv is clang7 normally on x86_64-darwin

Reply:
Oh never mind, I think I see the issue. qt5's internal mkDerivation uses clang5 for some reason. Should be as simple as this:

```diff
--- a/pkgs/misc/emulators/mame/default.nix
+++ b/pkgs/misc/emulators/mame/default.nix
@@ -33,7 +33,7 @@ let
 
   dest = "$out/opt/mame";
 in
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "mame";
   version = "0.237";
 
```